Mergin/validation.py

```diff
@@ -380,9 +389,29 @@ def check_svgs_embedded(self):
                     self.issues.append(SingleLayerWarning(lid, Warning.SVG_NOT_EMBEDDED))
                     break
 
-
-def warning_display_string(warning_id):
-    """Returns a display string for a corresponing warning"""
+    def check_editor_perms(self):
+        if self.project_permission == "editor":
+            # editor cannot change specific files - QGS project file, mergin-config.json file (e.g. selective sync changes)
+            for category in self.changes:
+                for file in self.changes[category]:
+                    path = file["path"]
+                    url = f"reset_file?layer={path}"
+                    if path.lower().endswith((".qgs", ".qgz")):
+                        self.issues.append(MultipleLayersWarning(Warning.EDITOR_PROJECT_FILE_CHANGE, url))
+                    elif path.lower().endswith("mergin-config.json"):
+                        self.issues.append(MultipleLayersWarning(Warning.JSON_CONFIG_CHANGE, url))
+            # editor cannot update nor delete a versioned file (e.g. .gpkg) or do non diff-based change (e.g. schema change)
+            for file in self.changes["removed"] or self.changes["updated"]:
```

Comment:
This `or`-thing does not work as expected
```suggestion
            for file in self.changes["removed"] or self.changes["updated"]:
```